From 9ab3fefd9b86c567ba6b0ea1429ce932572040c1 Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Tue, 26 Apr 2011 13:25:26 -0400
Subject: [PATCH 5/7] Add more machine type defines.

Add machine type defines for i386, arm/thumb, ia64, ebc, and x86_64.
---
 inc/ia32/pe.h   |    4 ++++
 inc/ia64/pe.h   |    4 ++++
 inc/x86_64/pe.h |    4 ++++
 3 files changed, 12 insertions(+)

diff --git a/inc/ia32/pe.h b/inc/ia32/pe.h
index 16e40ef..979b936 100644
--- a/inc/ia32/pe.h
+++ b/inc/ia32/pe.h
@@ -98,8 +98,12 @@ typedef struct _IMAGE_FILE_HEADER {
 #define IMAGE_FILE_MACHINE_R3000             0x162   // MIPS little-endian, 0540 big-endian
 #define IMAGE_FILE_MACHINE_R4000             0x166   // MIPS little-endian
 #define IMAGE_FILE_MACHINE_ALPHA             0x184   // Alpha_AXP
+#define IMAGE_FILE_MACHINE_ARMTHUMB_MIXED    0x1c2   // Arm/Thumb
 #define IMAGE_FILE_MACHINE_POWERPC           0x1F0   // IBM PowerPC Little-Endian
+#define IMAGE_FILE_MACHINE_IA64              0x200   // IA-64
 #define IMAGE_FILE_MACHINE_TAHOE             0x7cc   // Intel EM machine
+#define IMAGE_FILE_MACHINE_EBC               0xebc   // EFI Byte Code
+#define IMAGE_FILE_MACHINE_X64               0x8664  // x86_64
 //
 // Directory format.
 //
diff --git a/inc/ia64/pe.h b/inc/ia64/pe.h
index f67128d..b1cade2 100644
--- a/inc/ia64/pe.h
+++ b/inc/ia64/pe.h
@@ -113,8 +113,12 @@ typedef struct _IMAGE_FILE_HEADER {
 #define IMAGE_FILE_MACHINE_R3000             0x162   // MIPS little-endian, 0540 big-endian
 #define IMAGE_FILE_MACHINE_R4000             0x166   // MIPS little-endian
 #define IMAGE_FILE_MACHINE_ALPHA             0x184   // Alpha_AXP
+#define IMAGE_FILE_MACHINE_ARMTHUMB_MIXED    0x1c2   // Arm/Thumb
 #define IMAGE_FILE_MACHINE_POWERPC           0x1F0   // IBM PowerPC Little-Endian
+#define IMAGE_FILE_MACHINE_IA64              0x200   // IA-64
 #define IMAGE_FILE_MACHINE_TAHOE             0x7cc   // Intel EM machine
+#define IMAGE_FILE_MACHINE_EBC               0xebc   // EFI Byte Code
+#define IMAGE_FILE_MACHINE_X64               0x8664  // x86_64
 //
 // Directory format.
 //
diff --git a/inc/x86_64/pe.h b/inc/x86_64/pe.h
index 16e40ef..979b936 100644
--- a/inc/x86_64/pe.h
+++ b/inc/x86_64/pe.h
@@ -98,8 +98,12 @@ typedef struct _IMAGE_FILE_HEADER {
 #define IMAGE_FILE_MACHINE_R3000             0x162   // MIPS little-endian, 0540 big-endian
 #define IMAGE_FILE_MACHINE_R4000             0x166   // MIPS little-endian
 #define IMAGE_FILE_MACHINE_ALPHA             0x184   // Alpha_AXP
+#define IMAGE_FILE_MACHINE_ARMTHUMB_MIXED    0x1c2   // Arm/Thumb
 #define IMAGE_FILE_MACHINE_POWERPC           0x1F0   // IBM PowerPC Little-Endian
+#define IMAGE_FILE_MACHINE_IA64              0x200   // IA-64
 #define IMAGE_FILE_MACHINE_TAHOE             0x7cc   // Intel EM machine
+#define IMAGE_FILE_MACHINE_EBC               0xebc   // EFI Byte Code
+#define IMAGE_FILE_MACHINE_X64               0x8664  // x86_64
 //
 // Directory format.
 //
-- 
1.7.10.4

